src/main/java/org/threadly/litesockets/ThreadedSocketExecuter.java

```diff
@@ -38,22 +51,43 @@
   private WriteRunner writer;
 
 
+  /**
+   * <p>This constructor creates its own {@link SingleThreadScheduler} Threadpool to use for client operations.  This is generally 
+   * not recommended unless you are not doing many socket connections/operations.  You should really use your own multiThreaded 
+   * thread pool.</p>
+   */
   public ThreadedSocketExecuter() {
     schedulerPool = new SingleThreadScheduler(new ConfigurableThreadFactory("SocketClientThread", false, true, Thread.currentThread().getPriority(), null, null));
-    clientDistributer = new KeyDistributedScheduler(schedulerPool);
+    clientDistributer = new KeyDistributedExecutor(schedulerPool);
   }
 
-  public ThreadedSocketExecuter(SchedulerServiceInterface ssi) {
-    schedulerPool = ssi;
-    clientDistributer = new KeyDistributedScheduler(schedulerPool);
+  /**
+   * <p>This is provided to allow people to use java's generic threadpool scheduler {@link ScheduledExecutorService} </p>
+   * 
+   * @param exec The {@link ScheduledExecutorService} to be used for client/server callbacks.
+   */
+  public ThreadedSocketExecuter(ScheduledExecutorService exec) {
+    schedulerPool = new ScheduledExecutorServiceWrapper(exec);
+    clientDistributer = new KeyDistributedExecutor(schedulerPool);
+  }
+  
+  /**
+   * <p>Here you can provide a {@link ScheduledExecutorService} for this {@link SocketExecuterInterface}.  This will be used
+   * on accept, read, and close callback events.</p>
+   * 
+   * @param exec the {@link ScheduledExecutorService} to be used for client/server callbacks.
+   */
+  public ThreadedSocketExecuter(SimpleSchedulerInterface exec) {
+    schedulerPool = exec;
+    clientDistributer = new KeyDistributedExecutor(schedulerPool);
   }
 
   @Override
   public void addClient(final Client client) {
     if(! client.isClosed() && client.getProtocol() == WireProtocol.TCP && isRunning()) {
-      client.setThreadExecuter(clientDistributer.getSubmitterForKey(client));
-      client.setSocketExecuter(this);
-      if(client.getChannel() != null) {
+      client.setClientsThreadExecutor(clientDistributer.getSubmitterForKey(client));
+      client.setClientsSocketExecuter(this);
+      if(client.getChannel() != null && client.getChannel().isConnected()) {
```

Comment:
Since anyone could implement a client yes, but technically no.  Mainly because once the channel is set it should never ever change.  I probably need better documentation around this.